core/base/src/TString.cxx

```diff
@@ -131,6 +131,9 @@ TString::TString(const char *cs, Ssiz_t n)
       Zero();
       return;
    }
+   if (std::string(cs, 0, n).find(std::string("\0") != std::string::npos)) {
```

Comment:
Let's avoid a conversion to `std::string` which comes at additional cost and possibly memory allocations.  Instead, we should check the char array at `cs` directly

Reply:
Okay